File: LevelAndroidSDK/sdk/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/constants/ReporterError.java
package com.theshopatvsp.levelandroidsdk.ble.model.constants;

/**
 * Created by andrco on 10/3/16.
 */
public enum ReporterError {
    NOERROR(0), ReporterEnabled(1), ReporterInstanceError(2), DependentDataTypeError(3), ReporterDataNotEmpty(4);

    ReporterError(int e) {
        this.errorCode = e;
    }

    private int errorCode;

    public int getErrorCode() {
        return errorCode;
    }

    public static ReporterError getByErrorCode(int code) {
        for (ReporterError e : values()) {
            if (e.getErrorCode() == code) {
                return e;
            }
        }

        return null;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/model/constants/ReporterError.swift
//
//  ReporterError.swift
//  LevelIOSSdkExample
//
//  Created by Andrew Cook on 11/4/16.
//  Copyright © 2016 TheShop. All rights reserved.
//

import Foundation

enum ReporterError: Int {
    case NoError = 0, ReporterEnabled = 1, ReporterInstanceError = 2, DependentDataTypeError = 3, ReporterDataNotEmpty = 4
}
